src/pond/lib/timeseries.js

```diff
@@ -232,18 +232,20 @@ class TimeSeries {
             return;
         }
 
+        const columnList = this.columns();
+
         let columns;
         if (e instanceof TimeEvent) {
-            columns = ["time", ...this.columns()];
+            columns = ["time", ...columnList];
         } else if (e instanceof TimeRangeEvent) {
-            columns = ["timerange", ...this.columns()];
+            columns = ["timerange", ...columnList];
         } else if (e instanceof IndexedEvent) {
-            columns = ["index", ...this.columns()];
+            columns = ["index", ...columnList];
         }
 
         const points = [];
         for (const e of this._collection.events()) {
-            points.push(e.toPoint(this.columns()));
+            points.push(e.toPoint(columnList));
```

Comment:
This is the main change, right?  Most everything else is fixing the environment or the tests, yeah?